cpp/src/request_am.cpp

```diff
@@ -302,7 +332,7 @@ ucs_status_t RequestAm::recvCallback(void* arg,
       return UCS_INPROGRESS;
     }
   } else {
-    std::shared_ptr<Buffer> buf = amData->_allocators.at(UCS_MEMORY_TYPE_HOST)(length);
+    buf = amData->_allocators.at(UCS_MEMORY_TYPE_HOST)(length);
```

Comment:
*question*: Should we check for failed allocations here too?

Reply:
Given this is only for eager messages (usually up to few KB), I think if we ever hit this here there are bigger problems to worry about. Nevertheless, checking doesn't hurt, done in https://github.com/rapidsai/ucxx/pull/315/commits/93de79bdf0b15ace94959d4a82e2a9b736f7b873 .